subsys/lorawan/lorawan.c

```diff
@@ -571,11 +644,32 @@ void lorawan_register_dr_changed_callback(void (*cb)(enum lorawan_datarate))
 
 int lorawan_start(void)
 {
+	static bool already_inited;
+
 	LoRaMacStatus_t status;
 	MibRequestConfirm_t mib_req;
 	GetPhyParams_t phy_params;
 	PhyParam_t phy_param;
 
+	if (!already_inited) {
```

Comment:
Why does this need to be guarded behind `already_inited`? Is `lorawan_start` now expected to be called multiple times?

Reply:
Judging from naming only, I would have guessed that it's at least not forbidden to do so. Also, APIs evolve, so this seemed like the most defensive option